components/server/src/gitlab/gitlab-repository-provider.ts

```diff
@@ -99,4 +99,18 @@ export class GitlabRepositoryProvider implements RepositoryProvider {
         // FIXME(janx): Not implemented yet
         return [];
     }
+
+    async hasReadAccess(user: User, owner: string, repo: string): Promise<boolean> {
+        try {
+            // If we can "see" a project we are allowed to read it
+            const api = await this.gitlab.create(user);
+            const response = await api.Projects.show(`${owner}/${repo}`);
+            if (GitLab.ApiError.is(response)) {
+                return false;
+            }
+            return true;
+        } catch (err) {
+            return false;
+        }
+    }
```

Comment:
FIXME: This also fails for public GitLab repositories when you have no connected GitLab identity:

```
Oh, no! Something went wrong!
operation not permitted: missing get permission on snapshot
Unknown Error: { "code": 403 }
```